api/v1alpha1/virtualmachine_types.go

```diff
@@ -211,6 +211,9 @@ type VirtualMachineStatus struct {
 	// Phase describes the current phase information of the VirtualMachine.
 	Phase VMStatusPhase `json:"phase"`
 
+	// Conditions describes the current condition information of the VirtualMachine.
+	Conditions []VirtualMachineCondition `json:"conditions,omitempty"`
```

Comment:
PodStatus and NodeStatus both does `patchStrategy:"merge" patchMergeKey:"type"` wondering if we need it here as well?